src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Intellisense.cs

```diff
@@ -204,10 +204,10 @@ protected static void TypeMatchPriority(DType type, IList<IntellisenseSuggestion
                 if (!suggestion.Type.IsUnknown && 
 
                     // Most type acceptance is straightforward
-                    (type.Accepts(suggestion.Type) ||
+                    (type.Accepts(suggestion.Type, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true) ||
 
                     // Option Set expected types should also include the option set base as a reccomendation.
-                    (suggestion.Type.IsOptionSet && type.Accepts(DType.CreateOptionSetValueType(suggestion.Type.OptionSetInfo)))))
+                    (suggestion.Type.IsOptionSet && type.Accepts(DType.CreateOptionSetValueType(suggestion.Type.OptionSetInfo), exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true))))
```

Comment:
>exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true [](http://example.com/codeflow?start=128&length=82)

I understand the reason to make it explicit, to ensure there is no strange ordering issue, but most of the time they are going to be the same values for these three params. Is there a way to simplify this?

exact should have a default value as true.
useLegacyDateTimeAccepts is only applicable to certain types, so it's strange this to be a required parameter.

Not part of this change but after this change, may be we should try to simplify this.

Reply:
Makes sense, we should clean it up later.